src/server/endpoints/watch-later-endpoint.js

```diff
@@ -0,0 +1,45 @@
+'use strict';
+
+const usersDb = require('../collections/users-db');
+const tokensDb = require('../collections/tokens-db');
+
+function updateErrorhandle(res, result) {
+  if (result === 'Upadate Failed') {
+    res.status(500).json({'error': 'Update Failed'});
+    return;
+  }
+  if (result === undefined) {
+    res.status(500).json({'error': 'Unable connect database'});
+    return;
+  }
+}
+
+function watchLater(req, res) {
+  let token = req.get('Authorization');
+  let userId;
+
+  if (token === undefined) {
+    res.status(400).json({'error': 'unauthorization'});
+    return;
+  }
+  tokensDb.getToken(token, (tokenInfos) => {
+    userId = tokenInfos.userId;
+    let videoId = req.params.videoId;
+
+    usersDb.findUserInfoById(userId, (userInfos) => {
+      if (userInfos.email === undefined) {
+        res.status(404).json({'error': 'not found'});
+        return;
+      }
+      userInfos.watchlater.push({'videoId': videoId});
```

Comment:
we should handle the case when the user has already added the video to the watch later list. If it is already in the list we have to remove, if it is not, then we can push. Now if you click twice it will be added twice. You can use a the [filter](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/filter) function to find out whether it is already added or not. If you want to remove, you can use the [splice](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/splice) function.